neuraxle/metaopt/random.py

```diff
@@ -21,27 +21,266 @@
+    Base class For validation wrappers.
+    It has a scoring function to calculate the score for the validation split.
+
+    .. seealso::
+        :class`ValidationSplitWrapper`,
+        :class`ValidationSplitWrapper`,
+        :class`KFoldCrossValidationWrapper`,
+        :class`AnchoredWalkForwardTimeSeriesCrossValidationWrapper`,
+        :class`WalkForwardTimeSeriesCrossValidationWrapper`
+
+    """
+
+    def __init__(self, scoring_function: Callable = r2_score):
+        """
+        Base class For validation wrappers.
+        It has a scoring function to calculate the score for the validation split.
+
+        :param scoring_function: scoring function with two arguments (y_true, y_pred)
+        :type scoring_function: Callable
+        """
         MetaStepMixin.__init__(self)
         BaseStep.__init__(self)
         self.scoring_function = scoring_function
+
+
+class ValidationSplitWrapper(BaseValidation):
+    """
+    Wrapper for validation split that calculates the score for the validation split.
+
+    .. code-block:: python
+
+        random_search = Pipeline([
+            RandomSearch(
+                ValidationSplitWrapper(
+                    Identity(),
+                    test_size=0.1
+                    scoring_function=mean_absolute_relative_error,
+                    run_validation_split_in_test_mode=False
+                )
+            )
+        ])
+
+    .. seealso::
+        :class`BaseValidation`
+        :class`BaseCrossValidationWrapper`
+        :class`RandomSearch`
+
+    """
+
+    def __init__(
+            self,
+            wrapped: BaseStep,
+            test_size: float,
+            scoring_function=r2_score,
+            run_validation_split_in_test_mode=True
+    ):
+        """
+        :param wrapped: wrapped step
+        :param test_size: ratio for test size between 0 and 1
+        :param scoring_function: scoring function with two arguments (y_true, y_pred)
+        """
+        MetaStepMixin.__init__(self, wrapped)
+        BaseStep.__init__(self)
+        self.run_validation_split_in_test_mode = run_validation_split_in_test_mode
+        self.test_size = test_size
+        self.scoring_function = scoring_function
+
+    def transform(self, data_inputs):
+        """
+        Transform given data inputs without splitting.
+
+        :param data_inputs: data inputs
+        :return: outputs
+        """
+        return self.wrapped.transform(data_inputs)
+
+    def handle_fit_transform(self, data_container: DataContainer, context: ExecutionContext) -> (
+    'BaseStep', DataContainer):
+        """
+        Fit Transform given data inputs without splitting.
+
+        :param context:
+        :param data_container: DataContainer
+        :type data_container: DataContainer
+        :type context: ExecutionContext
+        :return: outputs
+        """
+        new_self, _ = self.handle_fit(data_container, context)
+        data_container = self.handle_transform(data_container, context)
```

Comment:
This seems like a rounded corner. I feel this may cause some algorithms to double in computing time. I didn't investigate before commenting so I might be wrong but I'm 70% confident on this comment.